image_widget_crop.module

```diff
@@ -203,3 +204,236 @@ function image_widget_crop_library_info_alter(&$libraries, $extension) {
+  $old_values = $form_state->getValues();
+
+  // Get values from js.
+  $user_values = $form_state->getUserInput();
+
+  $crop_types = $user_values['crop_preview_wrapper'];
+
+  foreach ($crop_types as $crop_type_id => $crop_container) {
+    $old_properties = $old_values['crop_preview_wrapper'][$crop_type_id]['crop_container']['values'];
+    $new_properties = $user_values['crop_preview_wrapper'][$crop_type_id]['crop_container']['values'];
+
+    // If we don't have old values and not creating new crop then simply
+    // continue.
+    if ($old_properties['edit'] == FALSE && $new_properties['crop_applied'] == 0) {
+      continue;
+    }
+
+    $image_widget_crop_manager = \Drupal::service('image_widget_crop.manager');
+
+    /** @var \Drupal\crop\Entity\CropType $crop_type */
+    $crop_type = \Drupal::entityManager()
+      ->getStorage('crop_type')
+      ->loadByProperties(['id' => $crop_type_id]);
+
+    // If the crop type needed is disabled or delete.
+    if (!isset($crop_type[$crop_type_id]) || empty($crop_type[$crop_type_id])) {
+      drupal_set_message(t("The CropType ('@cropType') is not active or not defined. Please verify configuration of image style or ImageWidgetCrop formatter configuration", ['@cropType' => $crop_type['crop_type_name']]), 'error');
+      return;
+    }
+
+    if ($old_properties['edit'] == FALSE && $new_properties['crop_applied'] == 1) {
+      if (!isset($crop_type) && empty($crop_type)) {
+        drupal_set_message(t("The type of crop does not exist, please check the configuration of the CropType ('@cropType')", ['@cropType' => $crop_type->label()]), 'error');
+        return;
+      }
+      $image_widget_crop_manager->applyCrop($new_properties, $old_properties, $crop_type[$crop_type_id]);
+    }
+
+    // Get all imagesStyle used this crop_type.
+    // TODO this is currently not returning anything. Check.
+    $image_styles = $image_widget_crop_manager->getImageStylesByCrop($crop_type[$crop_type_id]);
+    $crops = $image_widget_crop_manager->loadImageStyleByCrop($image_styles, $crop_type[$crop_type_id], $old_properties['file-id']);
+
+    if ($old_properties['edit'] == TRUE) {
+      if ($new_properties['crop_applied'] == 1) {
+        if ($image_widget_crop_manager->cropHasChanged($new_properties, $old_properties)) {
+          $image_widget_crop_manager->updateCrop($new_properties, $old_properties, $crop_type[$crop_type_id]);
+        }
+      }
+      elseif ($new_properties['crop_applied'] == 0) {
+        $image_widget_crop_manager->deleteCrop($old_properties['file-uri'], $crop_type[$crop_type_id], $old_properties['file-id']);
+      }
+    }
+
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function image_widget_crop_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
+  // We only want crop ui on images.
+  if (substr($form_id, 0, 5) == 'file_' && (substr($form_id, -10, 10) == '_edit_form' || substr($form_id, -17, 17) == '_inline_edit_form')) {
```

Comment:
There should be a base form_ID that you can alter on. See EntityForm::getBaseFormId(). You can then do image_widget_crop_form_file_form_alter(). getFormObject() also has a getOperation() method. Then you can check that on edit. Should simplify this a lot.